### PR TITLE
[MAJOR] Implement task management

### LINKED ISSUE #30: As a user, I want to manage task
Epic #28 
US: 8
## Acceptance Criteria
- Be able to create a task
- Be able to assign a task to an employee
- Be able to link a task to a project (not mandatory to link task when creating a task)
- Be able to edit the assigned task
- Be able to remove an assigned employee
- Be able to delete a task (archived or delete permanently)

### PR DESCRIPTION
**Pull request by bowsong**
Closes #30

### CODE REVIEW

**Review by davidthai0387 — DISMISSED** [review]
just a few comments. bon travail

File: backend/src/main/dto/TaskDTOs.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ export interface TaskCreationDTO {
   deadlineDate: Date;
   createdDate: Date;
   modifiedDate: Date;
-  projectId: number;
+  projectId?: number;
```
Comment (huangs08) [inline backend/src/main/dto/TaskDTOs.ts:8]: Do we allow creating a task without assigning it to a project?

File: frontend/src/components/Task/CreateTask.tsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  getAssignedByTaskId,
+  updateAssignedByTaskId,
+} from '../../services/AssignedAPI';
+import { useHistory } from 'react-router';
+
+const ManageTask: React.FC<any> = ({ id, edit }) => {
+  const [created, setCreated] = useState<boolean>(false);
+  const [projectList, setProjectList] = useState<any>([]);
+  const [employeeList, setEmployeeList] = useState<string[]>([]);
+  //merci lauren
```
Comment (huangs08) [inline frontend/src/components/Task/CreateTask.tsx:38]: Remove comment

File: backend/src/main/dto/TaskDTOs.ts (hunk 1 above)
Comment (bowsong) [reply backend/src/main/dto/TaskDTOs.ts:8]: Yeah from the acceptance criteria, but currently the database doesn't allow it, so Yup marks it as required.

**Review by davidthai0387 — APPROVED** [review]

**Review by davidthai0387 — APPROVED** [review]

**Review by huangs08 — APPROVED** [review]

**Pull request merged**